### PR TITLE
Should add functions for "Get Invite" and "Delete Invite" routes, fix minor issue in getChannelMessages.

### PR DESCRIPTION
**Pull request by RedTechKnight**
This will add the extensions functions 'getInvite' and 'deleteInvite' to the SmallDData<T> type, which correspond to the routes https://discordapp.com/developers/docs/resources/invite#get-invite and https://discordapp.com/developers/docs/resources/invite#delete-invite respectively. 

It also adds a data class to be used for storing the response for those paths. This class also makes use of a throwaway definition of a Guild object to make parsing the response easier, it should be able to be replaced easily with a more complete definition later if needed.

### CODE REVIEW

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/Requests.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.github.princesslana.smalldkt.SmallDData
+import com.github.princesslana.smalldkt.http.queryString
+import com.github.princesslana.smalldkt.http.get
+import com.github.princesslana.smalldkt.http.delete
+import com.github.princesslana.smalldkt.type.invite.resources.InviteData
+
+suspend fun SmallDData<*>.getInvite(code: String,includeCount: Boolean) =
+smallD.get(
+  "/invites/$code?" + queryString{
```
Comment (CoreyShupe) [inline src/main/kotlin/com/github/princesslana/smalldkt/type/invite/Requests.kt]: The `queryString` adds the `?` for you, no need to have it here

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.github.princesslana.smalldkt.type.invite.resources
+
+import com.github.princesslana.smalldkt.type.user.user.User
+import kotlinx.serialization.*
+import com.github.princesslana.smalldkt.type.channel.resources.*
+import kotlinx.serialization.Serializable
+
+
+@Serializable
+data class GuildObjectStub(
```
Comment (CoreyShupe) [inline src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: Is there any way to name this better, maybe something like `InvitePartialGuild` or something to that nature. Due to it being a top level class I'd rather not clutter the namespace with something vague like this.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        val verification_level: Int,
+        val icon: String?,
+        val banner: String?,
+        val id: String,
+        val vanity_url_code: String?,
+        val description: String?
+)
+
+@Serializable
+data class InviteData(
```
Comment (CoreyShupe) [inline src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: Look at other conventions to see how I kind of want the resources to be made:
https://github.com/princesslana/smalldkt/blob/master/src/main/kotlin/com/github/princesslana/smalldkt/type/channel/resources/ChannelEntity.kt#L11-L52
This kind of thing is used to allow for segmenting pieces and helping events, once everything is written out if it's used once then it'll be collapsed. Also, the name for this is kind of odd, see the connection with docs and naming conventions in the Channel as it should be similar.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        val inviter: User,
+        val code: String,
+        val guild: GuildObjectStub,
```
Comment (CoreyShupe) [inline src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: This is meant to be optional according to the docs.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt (hunk 3 above)
Comment (CoreyShupe) [inline src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: Where is this kind of data class coming from in the documentation? It seems like a mix of a few or from field testing. Please use the official docs to write the data classes then we can move from there. For reference this object should probably be a mix with the object and metadata with the meta-data parts being all optional.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt (hunk 2 above)
Comment (CoreyShupe) [inline src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: I also think we should just link this as a regular guild until one of the fields doesn't exist then work from there. Doing field testing with this kind of makes it vague as the docs don't necessarily define partial ~~afaik~~. They don't define partial, I'm going to be adding in special classes and you can start where every field is optional. Like a PartialGuild will have all the fields of a Guild but all of them will be optional. That's the best definition I got from people about what partial is.

**Review by CoreyShupe — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/Requests.kt (hunk 1 above)
Comment (RedTechKnight) [reply src/main/kotlin/com/github/princesslana/smalldkt/type/invite/Requests.kt]: Ah, terribly sorry. Fixed it in one of my new commits.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt (hunk 2 above)
Comment (RedTechKnight) [reply src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: Moved it to its own file under "type/guild/resources/GuildEntity.kt" and renamed it to PartialGuild, hopefully fixes this issue.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt (hunk 3 above)
Comment (RedTechKnight) [reply src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: The class name has been changed to follow the project's naming conventions.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt (hunk 4 above)
Comment (RedTechKnight) [reply src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: Among the other changes, the guild field is now defined as optional in the type as well.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt (hunk 3 above)
Comment (RedTechKnight) [reply src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: This is now the "InviteImpl" class more or less, which consists of the elements from the Invite object defined in the docs with the elements from the InviteMetadata object, all of which are optional in the new definition.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt (hunk 2 above)
Comment (RedTechKnight) [reply src/main/kotlin/com/github/princesslana/smalldkt/type/invite/resources/InviteEntity.kt]: The class has now been redefined as a PartialGuild, although not all fields are marked as Optional.

File: src/main/kotlin/com/github/princesslana/smalldkt/type/guild/resources/GuildEntity.kt [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+interface PartialGuild{
+    val id: Snowflake?
+    val name: String?
+    val icon: Optional<String>?
+    val splash: Optional<String>?
+    val owner: Boolean?
+    val owner_id: Snowflake?
+    val permissions: Int?
+    val region: String?
+    val afk_channel_id: Optional<Snowflake>?
```
Comment (CoreyShupe) [inline src/main/kotlin/com/github/princesslana/smalldkt/type/guild/resources/GuildEntity.kt:45]: this should have a default value of Optional.absent()
all optional marked items in docs need default values, this goes for everything else in this PR

**Review by CoreyShupe — CHANGES_REQUESTED** [review]

**Review by CoreyShupe — APPROVED** [review]

**Comment by CoreyShupe** [comment]
I was being dumb, my bad

**Pull request merged**